src/org/zaproxy/zap/extension/httpsinfo/ExtensionHttpsInfo.java

```diff
@@ -120,21 +125,30 @@ public URL getURL() {
 		}
 	}
 
-	void showSslTlsInfo(String hostname) {
-		new Thread(new BackgroundThread(hostname)).start();
+	void showSslTlsInfo(String hostname, HttpMessage msg) {
+		new Thread(new BackgroundThread(hostname, msg)).start();
 	}
 
 	private class BackgroundThread implements Runnable {
 
 		private String servername;
+		private HttpMessage baseMessage;
 
-		public BackgroundThread(String servername) {
+		public BackgroundThread(String servername, HttpMessage msg) {
+			this.baseMessage = msg;
 			this.servername = servername;
 		}
 
 		@Override
 		public void run() {
-			final SSLServer mServer = new SSLServer(servername);
+			final SSLServer mServer;
+			try {
+				mServer = new SSLServer(servername);
+			} catch (IOException ioe) {
+				getView().showWarningDialog(Constant.messages.getString("httpsinfo.failure.dialog.message"));
+				LOGGER.info("Could not establish a SSL/TLS connection with the server.");
```

Comment:
The exception message could be logged too (gives more info to the problem, see other comment below).